xds/src/main/java/io/grpc/xds/OrcaServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.xds.data.orca.v3.OrcaLoadReport;
+import com.github.xds.service.orca.v3.OpenRcaServiceGrpc;
+import com.github.xds.service.orca.v3.OrcaLoadReportRequest;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.util.Durations;
+import io.grpc.SynchronizationContext;
+import io.grpc.stub.ServerCallStreamObserver;
+import io.grpc.stub.StreamObserver;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class OrcaServiceImpl extends OpenRcaServiceGrpc.OpenRcaServiceImplBase {
+  private static final Logger logger = Logger.getLogger(OrcaServiceImpl.class.getName());
+
+  private final SynchronizationContext syncContext = new SynchronizationContext(
+      new Thread.UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+          logger.log(Level.SEVERE, "Exception!" + e);
+        }
+      });
+  private final long minReportIntervalNanos;
+  private final ScheduledExecutorService timeService;
+  private volatile ConcurrentHashMap<String, Double> metricsData = new ConcurrentHashMap<>();
+  private volatile double cpuUtilization;
+  private volatile double memoryUtilization;
+  @VisibleForTesting
+  final List<OrcaClient> clients = new ArrayList<>();
```

Comment:
There could be a lot of clients in this list, which means we have to be careful with it. It seems like it is unused though. Remove it and replace it with just a counter?